File: backend/entity/CompanyRequiredSkill .go
package entity

import (
	"gorm.io/gorm"
)

type CompanyRequiredSkill struct {
	gorm.Model
	SkillID          uint          `json:"skill_id"`
	IntershipPostID uint          `json:"intership_post_id"`
	Skill            Skill         `gorm:"foreignKey:SkillID"`
	IntershipPost   IntershipPost `gorm:"foreignKey:IntershipPostID"`
}

File: frontend/src/interfaces/CompanyRequiredSkill.ts
import type { SkillInterface } from './Skill';
// import type { InternshipPostInterface } from './InternshipPost';

export interface CompanyRequiredSkillInterface {
  ID?: number;                   // จาก gorm.Model.ID
  SkillID: number;
  IntershipPostID: number;
  Skill?: SkillInterface;               // ความสัมพันธ์ optional
//   IntershipPost?: InternshipPostInterface;
  CreatedAt?: string;
  UpdatedAt?: string;
  DeletedAt?: string | null;
}